### PR TITLE
Add 'PseudoTerminal' to 'TestSupport'

### PR DESCRIPTION
**Pull request by JhonnyBillM**
Add 'PseudoTerminal' to 'TestSupport' so it can be shared across different test targets 
[ @aciidb0mb3r suggestion ]

### CODE REVIEW

File: Tests/UtilityTests/ProgressBarTests.swift [hunk 1]
```diff
@@ -12,46 +12,10 @@ import XCTest
 import Utility
 import SPMLibc
 @testable import Basic
+@testable import TestSupport
```
Comment (aciidb0mb3r) [inline Tests/UtilityTests/ProgressBarTests.swift]: We should make the class public instead of doing a testable import

Comment (JhonnyBillM) [reply Tests/UtilityTests/ProgressBarTests.swift]: Fixed. Thanks!

File: Tests/UtilityTests/ProgressBarTests.swift [hunk 2]
```diff
@@ -88,10 +52,10 @@ final class ProgressBarTests: XCTestCase {
         }
         thread.start()
         runProgressBar(bar)
-        pty.closeSlave()
         // Make sure to read the complete output before checking it.
+        pty.close()
```
Comment (aciidb0mb3r) [inline Tests/UtilityTests/ProgressBarTests.swift]: We shouldn't close these fds at the same time. We should keep the separate closeSlave() and closeMaster() methods and change close() method to call those method in succession.

Comment (JhonnyBillM) [reply Tests/UtilityTests/ProgressBarTests.swift]: Oh I see. Fixed!

File: Tests/UtilityTests/ProgressBarTests.swift [hunk 3]
```diff
@@ -88,8 +52,8 @@ final class ProgressBarTests: XCTestCase {
         }
         thread.start()
         runProgressBar(bar)
-        pty.closeSlave()
         // Make sure to read the complete output before checking it.
+        pty.closeSlave()
```
Comment (aciidb0mb3r) [inline Tests/UtilityTests/ProgressBarTests.swift]: This diff is no longer required, right?

Comment (JhonnyBillM) [reply Tests/UtilityTests/ProgressBarTests.swift]: Ugh sorry, my bad checking. Fixed!

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Pull request merged**